fix protocol version

shubham3121 opened the pull request:
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

shubham3121 committed:
fix protocol version

eelcovdw commented on packages/syft/src/syft/service/job/job_stash.py:
```diff
@@ -607,6 +639,16 @@ def upgrade_job_v1_to_v2() -> list[Callable]:
     return [make_set_default("job_pid", None)]
 
 
+@migrate(JobInfoV1, JobInfo)
```
Not sure if we need a V2 + migrations here, V1 has a result as well.

shubham3121 replied:
Oh right, my bad we can remove migration

eelcovdw commented on packages/syft/src/syft/service/request/request.py:
```diff
@@ -1143,7 +1162,7 @@ class UserCodeStatusChange(Change):
     @property
     def code(self) -> UserCode:
-        return self.linked_user_code.resolve
+        return self.linked_obj.resolve
```
UserCodeStatusChange.linked_obj is now a UserCodeStatusCollection, StatusChange.linked_user_code is the old UserCode linked_obj. This should be `self.linked_user_code.resolve`

Is it possible to rename + drop with our current migrations?

shubham3121 replied:
That's my bad, will revert it to linked_user_code.
And yes its possible to rename, we have a `rename` transform. So in the new versionUserCodeStatusChange, linked_obj is no longer used and being renamed to `linked_user_code ` ?

eelcovdw replied:
Both are used, we decoupled code from status, and now mutate the UserCodeStatusCollection instead of the UserCode in `_run`. 

Probably can refactor it later to not link to the UserCode and only to the UserCodeStatusCollection, but decided to keep both in for now

shubham3121 committed:
remove migrations for JobInfo
- fix reference of linked_user_code in UserCodeStatus
- fix transform for user_code_from_code_status

eelcovdw committed:
fix integration tests

eelcovdw committed:
mypy fixes

eelcovdw committed:
fix mypy

eelcovdw committed:
merge dev

koenvanderveen committed:
skip test if windows
